db/db_impl.h

```diff
@@ -1192,7 +1192,7 @@ class DBImpl : public DB {
   std::string db_absolute_path_;
 
   // The options to access storage files
-  const EnvOptions env_options_;
+  EnvOptions env_options_;
```

Comment:
some clients store reference to this object. E.g.,

- `VersionSet`: https://github.com/facebook/rocksdb/blob/master/db/version_set.h#L849
- `LevelFileIteratorState`: https://github.com/facebook/rocksdb/blob/master/db/version_set.cc#L540

The second one's particularly scary because it accesses outside the lock during `NewSecondaryIterator`.

Reply:
want to check whether tsan catches the same thing?